src/test/util.ts

```diff
@@ -144,4 +144,13 @@ export function sanitizeForFilePath(s: string) {
   return s.replace(/[\x00-\x2F\x3A-\x40\x5B-\x60\x7B-\x7F]+/g, '-');
 }
 
+/**
+ * Get absolute path contained within parent directory, otherwise returns parent path
```

Comment:
Why don't we throw a nice error instead of constructing an unexpected path? I think that would be appreciated by the user.

Reply:
Wasn't sure what the protocol was for errors but I like the idea. Let me know what you think of df1e578